src/nvim/eval.c

```diff
@@ -8537,7 +8537,6 @@ static void f_getenv(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     rettv->vval.v_number = kSpecialVarNull;
     return;
   }
-  p = vim_strsave(p);
   rettv->vval.v_string = p;
```

Comment:
I think a copy is needed. Because otherwise VimL will free a `os_getenv` result, which is unexpected by `os_setenv` etc.

Reply:
It is a copy, isn't it?
`vim_getenv` does not use `mustfree`, and therefore it always needs to be freed / is allocated when not NULL (AFAICS).